File: src/ReactJSDotNet/Models/ToDo.cs
namespace ReactJSDotNet.Models
{
    public class ToDo
    {
        public ToDo()
        {

        }

        public ToDo(int id, string note, bool done)
        {
            Id = id;
            Note = note;
            Done = done;
        }

        public int Id { get; set; }
        public string Note { get; set; }
        public bool Done { get; set; }
    }
}

File: src/ReactJSDotNet/wwwroot/js/components/todo.js
'use strict';

export default ({ todo, onCheck, onDelete }) =>
    <li className="todo-item list-group-item">
        <div className="row">
            <div className="col-md-1">
                <input type="checkbox" checked={todo
            .done} onChange={() => onCheck({ id: todo.id, note: todo.note, done: !todo.done })}/>
            </div>
            <div className={'col-md-9' + (todo.done ? ' done' : '')}>
                {todo.note}
            </div>
            <div className="col-md-2">
                <button type="button" onClick={() => onDelete(todo)}>Remove</button>
            </div>
        </div>
    </li>;
